src/store/api/worldUsers.ts

```diff
@@ -0,0 +1,231 @@
+
+          updateCachedData((draft) => {
+            // We use splice here to remove all elements from the array and return them for processing
+            queuedChanges.splice(0).forEach(processUserDocChange(draft));
+          });
+        };
+
+        // TODO: move this interval into a proper const/config value somewhere
+        const processQueuedChangesIntervalId = setInterval(
+          processQueuedChanges,
+          5000
+        );
+
+        const worldUsersQuery = firebase
+          .firestore()
+          .collection("users")
+          .where("enteredVenueIds", "array-contains-any", relatedLocationIds);
+
+        const unsubscribeListener = worldUsersQuery.onSnapshot((snapshot) => {
+          // TODO: implement this properly + cleanup this placeholder stuff
+          const added = snapshot
+            .docChanges()
+            .filter((docChange) => docChange.type === "added").length;
+
+          const removed = snapshot
+            .docChanges()
+            .filter((docChange) => docChange.type === "removed").length;
+
+          const modified = snapshot
+            .docChanges()
+            .filter((docChange) => docChange.type === "modified").length;
+
+          console.log(
+            "worldUsersApi::worldUsers::snapshot",
+            "\n  snapshot.size=",
+            snapshot.size,
+            "\n  snapshot.docs.length = ",
+            snapshot.docs.length,
+            "\n  snapshot.docChanges().length",
+            snapshot.docChanges().length,
+            "\n  snapshot.docChanges() added:",
+            added,
+            "\n  snapshot.docChanges() removed:",
+            removed,
+            "\n  snapshot.docChanges() modified:",
+            modified,
+            "\n  queuedChanges.length",
+            queuedChanges.length
+          );
+
+          snapshot.docChanges().forEach((change) => {
+            // TODO: check if this document relates to the current user, if so, update it immediately
+            //   Note that we will need to pass their userId in and/or read it from somehwere to make this work..
+            //   probably just the queryArgs for this whole api listener
+            //
+            // const currentUserId = "TODO";
+            //
+            // if (change.doc.id === currentUserId) {
+            //   updateCachedData((draft) => {
+            //     processUserDocChange(draft)(change);
+            //   });
+            // } else {
+            queuedChanges.push(change);
+            // }
+          });
+        });
+
+        // Wait until the data no longer needs to be kept in our redux cache
+        await cacheEntryRemoved;
+
+        // Unsubscribe the firestore query snapshot listener
+        unsubscribeListener();
+
+        // Clear the interval for processing the queued changes
+        clearInterval(processQueuedChangesIntervalId);
+
+        // Make sure we process any last remaining queued changes
+        processQueuedChanges();
```

Comment:
[note] Here we ensure that we clear the interval + drain any remaining items in the `queuedChanges` array by calling `processQueuedChanges` one last time.